app/src/main/java/org/fossasia/openevent/app/core/main/MainActivity.java

```diff
@@ -168,14 +176,21 @@ private void loadFragment(int navItemId) {
         if (lastSelectedNavItemId == navItemId)
             return;
 
+        if (!isDashboardActive)
+            getSupportFragmentManager().popBackStack();
+
         binding.navView.setCheckedItem(navItemId);
         lastSelectedNavItemId = navItemId;
 
         Fragment fragment;
+        FragmentTransaction ft = fragmentManager.beginTransaction();
+
         switch (navItemId) {
             case R.id.nav_dashboard:
                 fragment = EventDashboardFragment.newInstance(eventId);
-                break;
+                ft.replace(R.id.fragment_container, fragment).commit();
+                isDashboardActive = true;
+                return;
```

Comment:
Use break

Unconditionally set `isDashboardActive` to true before the switch

Outside of switch, check isDashBoardActive, and add or replace fragments accrodingly and commit together